### PR TITLE
Implement mobius function

### PR DESCRIPTION
**Pull request by itsAkshayDubey**
### **Describe your change:**

- [x] Add an algorithm?
- [ ] Documentation change?
- [ ] Fix a bug or typo in an existing algorithm?

#### References

[#3232](https://github.com/TheAlgorithms/Java/issues/3232)

### **Checklist:**

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new Java files are placed inside an existing directory.
- [x] All filenames are in all uppercase characters with no spaces or dashes.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: src/test/java/com/thealgorithms/maths/MobiusFunctionTest.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+		//given
+		int number = -1;
+		int expectedValue = 1;
+
+		//when
+		int actualValue = MobiusFunction.mobius(number);
+
+		//then
+		assertEquals(expectedValue, actualValue);
+	}
```
Comment (siriak) [inline src/test/java/com/thealgorithms/maths/MobiusFunctionTest.java:33]: As far as I can tell, the Moebius function is undefined for negative integers and zero

File: src/main/java/com/thealgorithms/maths/MobiusFunction.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+		}
+		
+		if(primeFactorCount % 2 == 0) {
+			//if primeFactorCount is even
+			return 1;
+		}
+		else {
+			//if primeFactorCount is odd
+			return -1;
+		}
```
Comment (siriak) [inline src/main/java/com/thealgorithms/maths/MobiusFunction.java:58]: You can use a ternary operator instead

**Review by siriak — CHANGES_REQUESTED** [review]

File: src/test/java/com/thealgorithms/maths/MobiusFunctionTest.java (hunk 1 above)
Comment (itsAkshayDubey) [reply src/test/java/com/thealgorithms/maths/MobiusFunctionTest.java]: Fixed this [here](https://github.com/TheAlgorithms/Java/pull/3241/commits/bf9d3571477c85eea5024e7080c3d8f4f4955da7),  throwing IllegalArgumentException when method param is less than or equal to zero.

File: src/main/java/com/thealgorithms/maths/MobiusFunction.java (hunk 2 above)
Comment (itsAkshayDubey) [reply src/main/java/com/thealgorithms/maths/MobiusFunction.java]: Done [here](https://github.com/TheAlgorithms/Java/pull/3241/commits/13f7faa6a92d174ec39bf3c5c2575afee3975ccf)

**Review by siriak — APPROVED** [review]
Looks good, thanks!

**Comment by itsAkshayDubey** [comment]
Closing #3232

**Pull request merged**